### PR TITLE
introduce grcp exceptions

### PR DESCRIPTION
**Pull request by halakaraki**

### CODE REVIEW

File: wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Square Inc.
```
Comment (oldergod) [inline wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt:2]: 2021?

File: wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import okio.IOException
+
+/*
+Next steps:
+use GrpcStatus field in GrpcException (and IOExceptions !)
+copyrights on open source files
+GrpcStatus in its own file
+test that uses grpcStatus of 10 and catches the right exception
+ */
```
Comment (oldergod) [inline wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt:27]:
```suggestion
// TODO:
//  use GrpcStatus field in GrpcException (and IOExceptions !)
//  copyrights on open source files
//  GrpcStatus in its own file
//  test that uses grpcStatus of 10 and catches the right exception
```

**Review by oldergod — APPROVED** [review]

File: wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt (hunk 2 above)
Comment (swankjesse) [reply wire-library/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/GrpcException.kt:27]: Or just remove this, most of these are done!

File: wire-library/wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/internal/grpc.kt [hunk 3]
```diff
@@ -169,10 +171,14 @@ internal fun GrpcResponse.grpcResponseToException(suppressed: IOException? = nul
...
-    return IOException(
-        "gRPC call failure" +
-            " (HTTP status=$code, grpc-status=$grpcStatus, grpc-message=$grpcMessage)"
-    )
+    val grpcStatusInt = grpcStatus?.toIntOrNull()
+      ?: throw IOException(
+          "gRPC transport failure" +
+              " (HTTP status=$code, grpc-status=$grpcStatus, grpc-message=$grpcMessage)"
+      )
+    return when (grpcStatusInt) {
```
Comment (swankjesse) [inline wire-library/wire-grpc-client/src/jvmMain/kotlin/com/squareup/wire/internal/grpc.kt:179]: unnecessary when, can be just a return

**Review by swankjesse — APPROVED** [review]

**Pull request merged**